### PR TITLE
Add Elemental Resistance

### PR DESCRIPTION
**Pull request by h-yasha**

### CODE REVIEW

File: src/translate.js [hunk 1]
```diff
@@ -426,6 +426,16 @@ var multiLangData = {
...
         "zh": "Set sums of defense debuff values.The normal lower limit is 50%. Forfeit category is 60%.",
     },
+    "敵の耐性": {
+        "en": "Enemy Resistance",
+        "ja": "敵の耐性",
+        "zh": "Enemy Resistance",
+    },
+    "敵の耐性説明": {
+        "en": "Enemy's damage reduction for elements they are not weak to. Unite and Fight's NIGHTMARE(50%) and likes-",
+        "ja": "弱属性以外の攻撃に対して耐性を持つ点。星の古戦場HELL(50%)など。",
```
Comment (kei-gbf) [inline src/translate.js]: 持つ**点** -> 持つ**敵**
point -> enemy

File: src/global_logic.js [hunk 2]
```diff
@@ -601,8 +601,8 @@ module.exports.calcBasedOneSummon = function (summonind, prof, buff, totals) {
             var criticalRatio = 1.0
         }
         //Apply Elemental Resistance if not superior element.
-        damageUP -= totals[key]["typeBonus"] == 1.5 ? 0 : parseFloat(prof.enemyResistance);
-        
+        damageUP -= totals[key]["typeBonus"] == 1.5 ? 0 : Math.max(1, 0.01 * parseFloat(prof.enemyResistance));
```
Comment (kei-gbf) [inline src/global_logic.js]: can you check here, I am just reading code now.
- [ ] Math.max is for limit **minimum** value, is it always 1? (not tested)
- [ ] any cases enemyResistance is over the range
  - [x] it's limit min=0 max=100 in FormControl
  - [ ] load from save data may have illegal value

**Comment by kei-gbf** [comment]
@h-yasha 
Can you add display skill amount, it's useful for debug.

info-line

In `result.js` search `// Create buff info line`

**Comment by kei-gbf** [comment]
I found now it's implemented as add to "damageUp"
we can see damageUp filed for debug this, no needed the buff info line.

File: src/translate.js (hunk 1 above)
Comment (kagekiyo7) [reply src/translate.js]: proposal:

431 敵の耐性 → 非有利属性耐性

436 "ja": "非有利属性時の敵の耐性を設定します。古戦場HELL(50%)など。",

File: src/global_logic.js (hunk 2 above)
Comment (h-yasha) [reply src/global_logic.js]: it should've been Math.min.
i'll add both sides of acceptable range limit for it here too

**Comment by h-yasha** [comment]
**could not find any confirmation. was just speculation.**
Logic to be Updated. based on:
Testing with Shiva, in trial 
image
image

**Elemental Resistance is unique compared to final damage multipliers (seraphic, _which are irrelevant in this case_ /Shiva passive)**

File: src/profile.js [hunk 3]
```diff
@@ -858,6 +859,16 @@ var Profile = CreateClass({
                         </tr>
                     </TextWithTooltip>
 
+                    <TextWithTooltip tooltip={intl.translate("敵非有利耐性説明", locale)} id={"tooltip-enemy-resistance-detail"}>
```
Comment (kagekiyo7) [inline src/profile.js]: I think you should add step="5"
min="0" step="5" max="100"

> onbBlur={this.state.handleOnBlur} 

it's typo.

File: src/profile.js [hunk 4]
```diff
@@ -858,6 +859,16 @@ var Profile = CreateClass({
...
+                        <tr>
+                            <th className="bg-primary">{intl.translate("敵非有利耐性", locale)}</th>
+                            <td><InputGroup><FormControl type="number" min="0" max="100" value={this.state.enemyResistance}
+                                             onbBlur={this.state.handleOnBlur} onChange={this.handleSelectEvent.bind(this, "enemyResistance")}/>
```
Comment (kagekiyo7) [inline src/profile.js]: I think you should add step="5"
min="0" step="5" max="100"

> onbBlur={this.state.handleOnBlur} 

it's typo.

**Comment by h-yasha** [comment]
welp. I missed up.

**Pull request closed without merging**